python_modules/dagster/dagster_tests/definitions_tests/test_sensor_invocation.py

```diff
@@ -158,6 +159,123 @@ def my_cm_resource(_) -> Iterator[str]:
+            return RunRequest(
+                run_key=context.cursor, run_config={"foo": my_resource.a_str}, tags={}
+            )
+
+    @repository
+    def my_repo():
+        return [asset_a, asset_b, a_and_b_sensor]
+
+    with instance_for_test() as instance:
+        materialize([asset_a, asset_b], instance=instance)
+        ctx = build_multi_asset_sensor_context(
+            monitored_assets=[AssetKey("asset_a"), AssetKey("asset_b")],
+            instance=instance,
+            repository_def=my_repo,
+            resource_defs={"my_resource": MyResource(a_str="bar")},
+        )
+        assert cast(RunRequest, a_and_b_sensor(ctx)).run_config == {"foo": "bar"}
+
+
+def test_freshness_policy_sensor_invocation_resources() -> None:
+    class MyResource(ConfigurableResource):
+        a_str: str
+
+    @freshness_policy_sensor(asset_selection=AssetSelection.all())
+    def freshness_sensor(context, my_resource: MyResource) -> None:
+        assert context.minutes_late == 10
+        assert context.previous_minutes_late is None
+        assert my_resource.a_str == "bar"
+
+    with build_resources({"my_resource": MyResource(a_str="bar")}) as resources:
+        context = build_freshness_policy_sensor_context(
+            sensor_name="status_sensor",
+            asset_key=AssetKey("a"),
+            freshness_policy=FreshnessPolicy(maximum_lag_minutes=30),
+            minutes_late=10,
+            # This is a bit gross right now, but FressnessPolicySensorContext is not a subclass of
+            # SensorEvaluationContext and isn't set up to be a context manager
+            # Direct invocation of freshness policy sensors should be rare anyway
+            resources=resources,
+        )
+
+        freshness_sensor(context)
+
+
+def test_run_status_sensor_invocation_resources() -> None:
+    class MyResource(ConfigurableResource):
+        a_str: str
+
+    @run_status_sensor(run_status=DagsterRunStatus.SUCCESS)
+    def status_sensor(context, my_resource: MyResource):
+        assert context.dagster_event.event_type_value == "PIPELINE_SUCCESS"
+        assert my_resource.a_str == "bar"
+
+    @run_status_sensor(run_status=DagsterRunStatus.SUCCESS)
+    def status_sensor_no_context(my_resource: MyResource):
+        assert my_resource.a_str == "bar"
+
+    @op
+    def succeeds():
+        return 1
+
+    @job
+    def my_job_2():
+        succeeds()
+
+    instance = DagsterInstance.ephemeral()
+    result = my_job_2.execute_in_process(instance=instance, raise_on_error=False)
+
+    dagster_run = result.dagster_run
+    dagster_event = result.get_job_success_event()
+
+    context = build_run_status_sensor_context(
+        sensor_name="status_sensor",
+        dagster_instance=instance,
+        dagster_run=dagster_run,
+        dagster_event=dagster_event,
+        context=build_sensor_context(resource_defs={"my_resource": MyResource(a_str="bar")}),
+    )
+
+    status_sensor(context)
+    status_sensor_no_context(context)
```

Comment:
this feels off to me -- why do we invoke it like this instead of:

```
status_sensor_no_context(my_resource=MyResource(a_str="bar"))
```

As a user, it would be very hard to figure out that I had to invoke build_run_status_sensor_context and build_sensor_context in order to pass my resources through to the underlying function

Reply:
Simplified to just require `build_run_status_sensor_context` (nixing the weird nested context requirement). See my other comment about directly passing resources - I think that's the right impulse and something I can handle in a follow up.

Reply:
when that change lands, would we remove the ability to invoke sensors this way? my inclination would be that we shouldn't let people invoke sensors that have no context param with `build_..._context` in that future world, which makes me tempted to either not release this without the companion change, or live with these sensors not being directly-invokable until that change lands.

Reply:
@benpankow is the change to allow for direct resource passing particularly challenging. I'm inclined to agree wiht @OwenKephart that we should include that if possible.